File: payment_slips.R
# ==========================================================
# HIGHRIDGE CONSTRUCTION COMPANY
# WEEKLY PAYSLIP GENERATOR
# ==========================================================
# Author: Wadzanai Gwira
# Description:
# Generates weekly payslips for 400 workers and assigns
# employee levels based on salary and gender conditions.
# ==========================================================

set.seed(123)

tryCatch({
  
  # --------------------------------------------------------
  # STEP 1: CREATE 400 WORKERS DYNAMICALLY
  # --------------------------------------------------------
  
  workers <- data.frame(
    employee_id = 1:400,
    employee_name = paste("Worker", 1:400, sep = "_"),
    gender = sample(c("Male", "Female"), 400, replace = TRUE),
    salary = sample(5000:35000, 400, replace = TRUE)
  )
  
  cat("\n")
  cat("======================================================================\n")
  cat("          HIGHRIDGE CONSTRUCTION COMPANY\n")
  cat("          WEEKLY PAYROLL PROCESSING SYSTEM\n")
  cat("======================================================================\n")
  
  # --------------------------------------------------------
  # STEP 2: GENERATE PAYMENT SLIPS
  # --------------------------------------------------------
  
  for(i in 1:nrow(workers)) {
    
    employee_level <- "Unassigned"
    
    # ------------------------------------------------------
    # ASSIGN EMPLOYEE LEVELS
    # ------------------------------------------------------
    
    if(workers$salary[i] > 10000 &&
       workers$salary[i] < 20000) {
      
      employee_level <- "A1"
    }
    
    if(workers$salary[i] > 7500 &&
       workers$salary[i] < 30000 &&
       workers$gender[i] == "Female") {
      
      employee_level <- "A5-F"
    }
    
    # ------------------------------------------------------
    # GENERATE PAYSLIP
    # ------------------------------------------------------
    
    cat("\n")
    cat("======================================================================\n")
    cat("             HIGHRIDGE CONSTRUCTION COMPANY\n")
    cat("                    WEEKLY PAYSLIP\n")
    cat("======================================================================\n")
    
    cat("Pay Date         :", Sys.Date(), "\n")
    cat("Department       : Construction Operations\n")
    cat("Payment Type     : Weekly Salary\n")
    
    cat("----------------------------------------------------------------------\n")
    
    cat("Employee ID      :", workers$employee_id[i], "\n")
    cat("Employee Name    :", workers$employee_name[i], "\n")
    cat("Gender           :", workers$gender[i], "\n")
    cat("Gross Salary     : $",
        format(workers$salary[i], big.mark = ","), "\n")
    cat("Employee Level   :", employee_level, "\n")
    
    cat("----------------------------------------------------------------------\n")
    
    cat("Payment Status   : APPROVED\n")
    cat("Prepared By      : Payroll Department\n")
    
    cat("======================================================================\n")
  }
  
},
error = function(e) {
  cat("An error occurred:", e$message)
})

File: payment_slips.py
# ==========================================================
# HIGHRIDGE CONSTRUCTION COMPANY
# WEEKLY PAYSLIP GENERATOR
# ==========================================================
# Author: Wadzai Gwira
# Description:
# Generates weekly payslips for 400 workers and assigns
# employee levels based on salary and gender conditions.
# ==========================================================

import random
from datetime import date

try:

    # ------------------------------------------------------
    # STEP 1: CREATE 400 WORKERS DYNAMICALLY
    # ------------------------------------------------------
    workers = []

    for i in range(1, 401):

        worker = {
            "employee_id": i,
            "employee_name": f"Worker_{i}",
            "gender": random.choice(["Male", "Female"]),
            "salary": random.randint(5000, 35000)
        }

        workers.append(worker)

    # ------------------------------------------------------
    # STEP 2: GENERATE PAYMENT SLIPS
    # ------------------------------------------------------
    print("\n")
    print("=" * 70)
    print("         HIGHRIDGE CONSTRUCTION COMPANY")
    print("         WEEKLY PAYROLL PROCESSING SYSTEM")
    print("=" * 70)

    for worker in workers:

        employee_level = "Unassigned"

        # --------------------------------------------------
        # ASSIGN EMPLOYEE LEVELS
        # --------------------------------------------------

        # Condition 1
        if worker["salary"] > 10000 and worker["salary"] < 20000:
            employee_level = "A1"

        # Condition 2
        if (
            worker["salary"] > 7500
            and worker["salary"] < 30000
            and worker["gender"] == "Female"
        ):
            employee_level = "A5-F"

        # --------------------------------------------------
        # GENERATE PAYSLIP
        # --------------------------------------------------

        print("\n")
        print("=" * 70)
        print("             HIGHRIDGE CONSTRUCTION COMPANY")
        print("                    WEEKLY PAYSLIP")
        print("=" * 70)

        print(f"Pay Date         : {date.today()}")
        print("Department       : Construction Operations")
        print("Payment Type     : Weekly Salary")

        print("-" * 70)

        print(f"Employee ID      : {worker['employee_id']}")
        print(f"Employee Name    : {worker['employee_name']}")
        print(f"Gender           : {worker['gender']}")
        print(f"Gross Salary     : ${worker['salary']:,.2f}")
        print(f"Employee Level   : {employee_level}")

        print("-" * 70)

        print("Payment Status   : APPROVED")
        print("Prepared By      : Payroll Department")

        print("=" * 70)

except ValueError as ve:
    print("Value Error:", ve)

except KeyError as ke:
    print("Missing Key:", ke)

except TypeError as te:
    print("Type Error:", te)

except Exception as e:
    print("An unexpected error occurred:", e)
